Added PieChart

rickimaru18 opened the pull request:
# 📲 What #

* Added new chart: PieChart (NORMAL).
* Updated LineChart to support "title" display.

# 🤔 Why #

* New feature

# 🛠 How #

* Added PieChart composable
  + Included composable preview
  + Updated PieChart sample fragment
* Added "title" property in LineChart
* Updated chart template to support title display

# ✅ Acceptance criteria #

- [x] Composable preview added
- [x] Added UI Tests
- [x] Passing Unit Tests
- [x] Passing UI Tests

# ⏰ TODO #

* None

# 🔗 References #

* URLS or Images or Ticket links.
<img width="305" alt="image" src="https://user-images.githubusercontent.com/13028267/115530195-21400480-a2c6-11eb-94a5-cdca45b3bab5.png">

rickimaru18 committed:
PieChart initial version

rickimaru18 committed:
Update PieChart and LineChart

# In chart template
- Added title support.
- Fix vertical legends layout.

# LineChart update
- Added "title" property in LineChart and display it in sample fragment.
- Updated LineChart's instrumented test to test title display.

# PieChart update
- Update sample fragment to be the same with LineChart.
- Added PieSeries which is the PieChart's model.
- In PieChart...
  + Follow specs v0.0.1 properties.
  + Segregate drawing functions.
  + Added preview composable.
  + Added instrumented test.

# In ChartUtils, added end angle calculation which is used by PieChart
- Added unit test for "getEndAngles".

rickimaru18 committed:
Update gradle version and fix build warnings

# Update gradle versions for Android Studio Canary update (v13 --> v14)

# Fix build warnings
- Removed unused variables
- Use unused variable in PieChart

rexendy commented on kwarchart/src/main/java/com/kwarchart/android/chart/PieChart.kt:
```diff
@@ -1,15 +1,119 @@
+                    Log.d("SHIT", "value = ${chartData[i].value}")
+                    Log.d("SHIT", "endAngle = $endAngle  :  x = $x  :  y = $y")
+
+
```
remove logs and spaces

rexendy commented on kwarchart/src/main/java/com/kwarchart/android/chart/PieChart.kt:
```diff
@@ -1,15 +1,119 @@
+                        majorAxis + ((startXY.x - majorAxis - x) / 2f),
+                        -((minorAxis - (minorAxis + y)) / 2f)
+                    )
+//                    if (i == 0)
```
remove if not used

rexendy commented on sample/src/main/res/layout/fragment_pie.xml:
```diff
@@ -6,17 +6,13 @@
-    <TextView
-        android:id="@+id/text_notifications"
+    <androidx.compose.ui.platform.ComposeView
+        android:id="@+id/chartPie"
```
for xml pls use snake case and for classes/activiites/fragment pls use camelCase

* to determine that this is from xml layout

rexendy commented on kwarchart/src/main/java/com/kwarchart/android/chart/PieChart.kt:
```diff
@@ -1,119 +1,171 @@
-                startAngle += endAngle
-            }
+            drawArcs(data)
+//            drawArcValues(data)
```
remove comment if not needed

rexendy commented on sample/src/main/java/com/kwarchart/sample/ui/line/LineFragment.kt:
```diff
@@ -53,6 +53,7 @@ class LineFragment : Fragment() {
                             lineViewModel.spentSeries.value!!,
                             lineViewModel.goalSeries
                         ),
+                        title = "This week's transactions",
                         yAxisName = "Spent",
                         xAxisName = "Day",
```
pls add the strings to strings.xml

rexendy commented on sample/src/main/java/com/kwarchart/sample/ui/pie/PieFragment.kt:
```diff
@@ -44,9 +51,9 @@ class PieFragment : Fragment() {
-                        legendY = "Savings",
-                        chartData = it
+                        data = it,
+                        title = "This month's transactions",
```
add the text to string.xml

rexendy commented on kwarchart/src/main/java/com/kwarchart/android/chart/PieChart.kt:
```diff
@@ -101,10 +128,10 @@ private fun <T> DrawScope.drawArcValues(data: List<PieSeries<T>>) {
-        this.size.width / 2f,
-        this.size.height / 2f
-    )
+//    val origin = Offset(
+//        this.size.width / 2f,
+//        this.size.height / 2f
+//    )
```
remove comment if not needed

rickimaru18 committed:
Merge branch 'develop' into feature/normal_pie_chart

rickimaru18 committed:
PieChart legend update

# Update PieChart UI test
- Create a PieChart with a label that contains the value.
- Update legend assertions in "pieSeriesLegendTest" and "pieSeriesNoLegendTest".

# Added TODO comment in PieChart.kt

# Update PieSeries legend in sample app
- Create a PieChart with a label that contains the value.

rickimaru18 committed:
IDs rename and use string resource

# Added new strings in strings.xml and use it in sample fragments
- Chart titles.
- Axes names.

# Use snake case for IDs in layout file

rexendy commented:
@rickimaru18 pls merge to develop branch thank you
